cargo-nextest/src/dispatch.rs

```diff
@@ -308,8 +318,17 @@ impl TestBuildFilter {
             &path_mapper,
             self.platform_filter.into(),
         )?;
-        let test_filter =
-            TestFilterBuilder::new(self.run_ignored, self.partition.clone(), &self.filter);
+        let exprs = self
+            .expr_filter
+            .iter()
+            .map(|input| FilteringExpr::parse(input, graph))
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Could this be reordered to happen before the build happens? This will cause parsing errors to be printed out immediately rather than after the build.